tests/components/recorder/test_init.py

```diff
@@ -92,6 +92,7 @@ async def test_shutdown_before_startup_finishes(hass):
     await hass.data[DATA_INSTANCE].async_db_ready
     await hass.async_block_till_done()
 
+    # This means the connection is checked out by some random thread, and never released
```

Comment:
Comment added to explain why the test fails, to be removed before the PR is merged